test/promql/test_network.yml

```diff
@@ -0,0 +1,108 @@
+# This is the main input for unit testing.
+# Only this file is passed as command line argument.
+
+rule_files:
+    - ../../prometheus/base/prometheus.rules.yaml
+
+evaluation_interval: 30s
+
+tests:
+ - interval: 30s
+   # Series data 
+   input_series:
```

Comment:
Looks good!

I think we should introduce inputs with missing data. i.e. `_` readings, and verify the results make sense.

I imagine that the way bugs will manifests for this sort of code is through readings that we did not expect to happen, and our way to recreate them will be generate inputs that behave this way.

We don't need to try and anticipate them all now, but the once we know of, we should already introduce (e.g. recordings for pods that are already dead/deleted).

Reply:
Yes, I intend to add both `_` and `0` to simulate restart of count.

Reply:
I updated the tests to include both `0` and `_`.  
I also added some explanations for the tests.